silk/tests/openthread/ot_test_meshcop_joiner_discerner.py

```diff
@@ -0,0 +1,177 @@
+        self.network_name = "SILK-{0:04X}".format(random.randint(0, 0xffff))
+        cmd = "form {}".format(self.network_name)
+        self.commissioner.wpanctl_async("form", cmd, "Successfully formed!", 20)
+        self.wait_for_completion(self.device_list)
+        leader_node_type = self.commissioner.wpanctl("get", "get " + wpan.WPAN_NODE_TYPE,
+                                                     2).split("=")[1].strip()[1:-1]
+        self.assertTrue(leader_node_type == "leader", "Leader is not created correctly!!!")
+
+    @testcase.test_method_decorator
+    def test02_verify_discerner_value_after_device_reset(self):
+        # Verify Discerner value after device reset on Joiner.
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == 0)
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == 0)
+
+    @testcase.test_method_decorator
+    def test03_set_joiner_discerner_on_joiner1_and_joiner2(self):
+        # Set Joiner Discerner on `joiner1` and `joiner2`.
+        self.joiner1.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, str(D_LEN1))
+        self.joiner1.set(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE, DISCERNER1)
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == D_LEN1)
+        verify(int(self.joiner1.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == int(DISCERNER1, 0))
+
+        self.joiner2.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, str(D_LEN2))
+        self.joiner2.set(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE, DISCERNER2)
+        verify(int(self.joiner2.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == D_LEN2)
+        verify(int(self.joiner2.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == int(DISCERNER2, 0))
+
+    @testcase.test_method_decorator
+    def test04_verify_clearing_of_previously_set_joiner_discerner(self):
+        # Set Joiner Discerner on `joiner3`.
+        self.joiner3.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, str(D_LEN2))
+        self.joiner3.set(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE, DISCERNER2)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == D_LEN2)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == int(DISCERNER2, 0))
+
+        # Clear Joiner Discerner on `joiner3`.
+        self.joiner3.set(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH, '0')
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_BIT_LENGTH), 0) == 0)
+        verify(int(self.joiner3.get(wpan.WPAN_THREAD_JOINER_DISCERNER_VALUE), 0) == 0)
+
+    @testcase.test_method_decorator
+    def test05_add_joiners_on_commissioner(self):
+        self.commissioner.commissioner_start()
+        self.wait_for_completion(self.device_list)
+
+        # Add `joiner1` and `joiner2` using joiner discerner.
+        self.commissioner.commissioner_add_joiner_with_discerner(DISCERNER1, D_LEN1, PSK1, JOINER_TIMOUT)
```

Comment:
please check if commissioner_add_joiner_with_discerner() has been added to wpantund_base.py

Reply:
Sure. Checked this function is present in the master.